src/main/java/com/mongodb/jdbc/MongoRunCmd.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.jdbc;
+
+import com.mongodb.jdbc.logging.MongoLogger;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.logging.Level;
+import org.bson.Document;
+
+public class MongoRunCmd {
+
+    private final MongoConnection connection;
+    private final MongoLogger logger;
+
+    /** Native method to send commands via JNI. */
+    public native byte[] runCommand(byte[] command, int length) throws SQLException;
+
+    /**
+     * Constructs a MongoRunCmd instance associated with a MongoConnection, used to obtain logger.
+     *
+     * @param connection The MongoConnection instance.
+     */
+    public MongoRunCmd(MongoConnection connection) {
+        this.connection = connection;
+        this.logger = connection.getLogger();
+    }
+
+    /**
+     * Executes a BSON command and returns the response as a Document.
+     *
+     * @param command The BSON command to execute.
+     * @return The response Document.
+     * @throws SQLException If the command execution fails.
+     */
+    public Document runCommand(Document command) throws SQLException {
```

Comment:
That's what I found too, the BsonDocument is strongly typed and provides more type safety.  I made the change to switch to using BsonDocument instead of Document.